### PR TITLE
Project todos - Linda Hintze

### PR DESCRIPTION
**Pull request by lindahz**
**Link to my deployed project:**
https://daily-planner-by-linda.netlify.app/

**My reflections on how this week's project turned out:**
It went well. I did not have time to complete two things that was on my todo list (lol): create a library for reusable components and filter on tasks that are uncompleted vs. all tasks. 

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,31 +4,13 @@
...
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>Technigo React App</title>
+    <title>Daily Planner by Linda Hintze</title>
+    <link rel="icon" href="https://api.iconify.design/emojione:purple-heart.svg" type="image/svg">
```
Comment (ssundblom) [inline code/public/index.html:8]: Love the favicon 💜

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+
+const store = createStore(
+  reducer, 
+  persistedState,
+  window.__REDUX_DEVTOOLS_EXTENSION__&& window.__REDUX_DEVTOOLS_EXTENSION__()
+)
+
+store.subscribe(() => {
+  localStorage.setItem("reduxState", JSON.stringify(store.getState()))
+})
```
Comment (ssundblom) [inline code/src/App.js:29]: So nice feature to have this local storage! Want to implement this on my own code!

File: code/src/components/AddTask.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+`
+
+const TaskButton = styled.button`
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+
+  &:disabled {
+    opacity: 0.4;
+  }
```
Comment (ssundblom) [inline code/src/components/AddTask.js:83]: This was so nice done! Think it looks super clean and goes well with the rest of the styling!

File: code/src/components/NoTasks.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+export const NoTasks = () => {
+  return (
+   <Container>
+     <Image
+      src={Logo}
+      alt="clouds"
+     />
+     <Title>
+       just relax, you have nothing to do right now. 
+     </Title>
```
Comment (ssundblom) [inline code/src/components/NoTasks.js:16]: This is just sooo cute!

File: code/src/components/TaskList.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+export const TaskList = () => {
+
+  const items = useSelector (
+    (store) => store.tasks.items
+  )
+
+  return (
+    <Section>
+      <Header />
+      {items.length === 0 ? <NoTasks /> :
```
Comment (ssundblom) [inline code/src/components/TaskList.js:21]: Very cool to handle an empty state 😍

File: code/src/reducers/tasks.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+export const tasks = createSlice({
+  name: 'tasks',
+  initialState: {
+    items: [
+      {id: 1606460400000, text: 'eat well and excercise', complete: false},
+      {id: 1606547700000, text: 'spend time with friends and family', complete: false},
+      {id: 1606548600000, text: 'express gratitude', complete: false},
+      {id: 1606463100000, text: 'laugh', complete: false}
+    ]
```
Comment (ssundblom) [inline code/src/reducers/tasks.js:11]: Clever to use time stamp as id!

File: code/src/components/CustomCheckbox.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+          checked={checked}
+          onChange={handleOnChange}
+        >
+        </Checkbox>
+      </label>
+    </div>
+  )
+}
+
+const HiddenCheckbox = styled.input.attrs({type: "checkbox"})`
```
Comment (ssundblom) [inline code/src/components/CustomCheckbox.js:39]: Ahh! The magic, hidden checkbox 🙈😅

**Review by ssundblom — CHANGES_REQUESTED** [review]
WOW! It really looks amazing Linda! I get more chill of just looking at it <3 
The color scheme is on point! Love the yogi that shows when the list is cleared 🧘‍♀️ 
Very nice that you have local storage! I will for sure implement this to my todo app 🙃
Good job with the styled components! 

There's one thing - when I do a *clear all* I can't seem to add another task again? I tried the app in an incognito window as well and it seems to work better then, but sometimes it kinda 'strikes' and don't want to add the task? 

Besides this you've built a wonderful, red level app! Really good work, I'm super impressed 😍

File: code/src/components/Task.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        isChecked={task.complete} 
+        onChangeHandler={handleCheckboxClick}
+      />
+      <TextContainer>
+        <TaskName>
+          {task.text}
+        </TaskName>
+        <Date>
+        created at {moment(task.id).format('LT')}
+      </Date>
```
Comment (HIPPIEKICK) [inline code/src/components/Task.js:36]: This shows only the time and not the date 🤔

File: code/src/components/AddTask.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    setNewTask('')
+  }
+
+  return (
+    <Container
+      onSubmit={handleSubmit}
+    >
+      <TaskButton
+        type="submit"
+        disabled={newTask.length < 5 ? true : false}
```
Comment (HIPPIEKICK) [inline code/src/components/AddTask.js:26]: What if I wanna "code" ;)

File: code/src/components/Task.js (hunk 8 above)
Comment (lindahz) [reply code/src/components/Task.js:36]: Yes, on purpose, which is why I used LT, and calling it a "daily planner" ;)

File: code/src/components/AddTask.js (hunk 9 above)
Comment (lindahz) [reply code/src/components/AddTask.js:26]: true.. 🙃

File: code/src/components/Task.js (hunk 8 above)
Comment (HIPPIEKICK) [reply code/src/components/Task.js:36]: Haha sorry, my comments are not always as clear as I think they are 🙈 Maybe call the styled component `Time` instead? 😇

**Pull request closed without merging**